zerver/lib/events.py

```diff
@@ -52,6 +51,22 @@ def avatar_url(userdict):
              'full_name': userdict['full_name']}
             for userdict in get_active_user_dicts_in_realm(user_profile.realm)]
 
+def do_update_buddy_list(user_profile, buddy_profile, should_add):
+    # type: (UserProfile, str, bool) -> None
+
+    if should_add:
+        BuddyList.objects.get_or_create(user=user_profile, buddy=buddy_profile)
+    else:
+        BuddyList.objects.filter(user=user_profile, buddy=buddy_profile).delete()
```

Comment:
This should probably do a `send_event` call to notify frontends about the changes; we probably want to design that in a later stage, though.  Maybe just add a comment?

The send_event logic will care whether the user was actually added or not, though, so I think it makes sense to have this actually check whether one existed or not, and at least do a `logging.warn` if it the previous state was inconsistent with the change (might not want to throw an error with an endpoint intended to be idempotent).